_python/main/sanitize.py

```diff
@@ -0,0 +1,140 @@
+"""
+
+## helpers ##
+
+def get_words(s):
+    return set(s.strip().split())
+
+
+## allow-lists ##
+
+@lru_cache()
+def get_allow_lists():
+    # tags created by ckeditor
+    allowed_tags = get_words("""
+        p br
+        ol ul li
+        blockquote
+        strong em u s sub sup
+        img
+        table caption thead th tbody tr td
+        hr
+    """)
+
+    # legacy tags
+    allowed_tags |= get_words("""
+        a address b blockquote br center cite col colgroup dd del div dl dt em h1 h2 h3 h4 h5 h6 header hr i img li mark ol 
+        p pre small span strike strong sub sup table tbody td th thead time tr u ul wbr
+    """)
+
+    # attributes created by ckeditor
+    # (overridden by the legacy definition, which is a super-set of this)
+    # allowed_attributes = {
+    #     'img': ['src', 'alt', 'style'],
+    #     'table': ['cellpadding', 'cellspacing', 'align', 'border', 'style'],
+    # }
+
+    # legacy attributes
+    allowed_attributes = {
+        'a': {'title', 'rel', 'name', 'class', 'id', 'style', 'href'},
+        'b': {'id'},
+        'blockquote': {'id', 'class', 'style'},
+        'br': {'class', 'style'},
+        'center': {'class', 'style'},
+        'col': {'width'},
+        'dd': {'style'},
+        'div': {'title', 'tabindex', 'dir', 'class', 'id', 'style'},
+        'dl': {'style'},
+        'em': {'class', 'style'},
+        'h1': {'id', 'dir', 'class', 'style'},
+        'h2': {'title', 'id', 'dir', 'class', 'style'},
+        'h3': {'id', 'class', 'style'},
+        'h4': {'class', 'style'},
+        'h5': {'id', 'style'},
+        'h6': {'class', 'style'},
+        'header': {'class'},
+        'hr': {'class', 'style'},
+        'img': {'align', 'alt', 'border', 'class', 'height', 'id', 'src', 'srcset', 'style', 'tabindex', 'title', 'width'},
+        'li': {'id', 'value', 'dir', 'class', 'style'},
+        'ol': {'type', 'start', 'class', 'style'},
+        'p': {'title', 'dir', 'class', 'id', 'lang', 'style'},
+        'pre': {'id', 'style'},
+        'span': {'title', 'id', 'lang', 'class', 'style'},
+        'strike': {'style'},
+        'strong': {'id', 'class', 'style'},
+        'sup': {'id', 'class', 'style'},
+        'table': {'align', 'bgcolor', 'border', 'cellpadding', 'cellspacing', 'class', 'id', 'style', 'summary', 'width'},
+        'td': {'align', 'class', 'colspan', 'headers', 'id', 'rowspan', 'style', 'valign', 'width'},
+        'th': {'id', 'scope'},
+        'time': {'datetime', 'class'},
+        'tr': {'class', 'style'},
+        'u': {'class', 'style'},
+        'ul': {'type', 'id', 'class', 'style'},
+    }
+
+    # styles created by ckeditor
+    allowed_styles = (
+        # ckeditor image tags
+        get_words("border-width border-style margin float width height")
+        # ckeditor table tags
+        | get_words("width height")
+    )
+
+    # legacy styles
+    allowed_styles |= get_words("""
```

Comment:
Wow. I had no idea there were all these nice operators for working with sets (`|` for union, `|=`, etc.) Huzzah!